### DISCUSSION TITLE
Is there a way to access the test summary directly?

### DISCUSSION THREAD

**Discussion by enrico-semrau** [behave/behave#960] [Q&A] [answered]
Hey guys, I have searched my a** off but couldn't manage to find a solution.
I work on a solution to send the test summary to slack after a run.
Ideally I could somehow access the test summary and store it in a file or variable to work with it later in the pipeline.

What I do now, because I am not able to access it, I am saving every failed scenario into a file and then print out the list / send it to slack. 

But if there is a way to avoid this whole implementation and easily use the summary then it would save a lot of time and code and would be highly appreciated.

So I only need this part here:
####
0 features passed, 1 failed, 121 skipped
0 scenarios passed, 1 failed, 329 skipped
0 steps passed, 0 failed, 6595 skipped, 0 undefined, 1 untested
Took 0m0.000s
####

Do you have any idea?

**Comment by jenisys** [comment, marked as answer]
It depends what you actually want to have:

* The list of failing scenarios or
* The summary statistics (counters per category: feature/scenario/step)

For the list of failing scenarios, use the `rerun` formatter and write it to a file.
It contains all the file locations of failed scenarios to simplify the rerun with `behave @rerun.txt` (with output filename=„rerun.txt“).
The normal „behave.ini“ already uses this functionality to write failing scenario file locations to „rerun.txt“ file.

HINT: You can have multiple formatters but only at most one that writes to stdout/console.
SEE ALSO:

* https://github.com/behave/behave/blob/master/behave.ini#L15 (and line 17)

**Reply by jenisys** [reply]
Otherwise, if you are only interested in the statistics:

* Use `behave | tail -4` to capture only the last four lines of the output (the „tail output“)
* Write another formatter that computes the Summary from the Summary reporter and use the `—output` option to direct to a file.

**Reply by enrico-semrau** [reply]
Thx a ton for your help.
the statistics in total would be great. Just have no clue on how to write a formatter :)

**Reply by jenisys** [reply]
Note that the `behave.summary` module provides now functionality to compute the statistics independent of the `summary` reporter.

SEE ALSO:
* [behave.summary.SummaryCollector](https://github.com/behave/behave/blob/main/behave/summary.py#L505)
